inc/auth.class.php

```diff
@@ -581,12 +581,16 @@ function Login($login_name, $login_password, $noauto=false) {
          if ($this->getAlternateAuthSystemsUserLogin($authtype)
              && !empty($this->user->fields['name'])) {
             // Used for log when login process failed
-            $login_name                     = $this->user->fields['name'];
-            $this->auth_succeded            = true;
-            $this->extauth                  = 1;
-            $this->user_present             = $this->user->getFromDBbyName(addslashes($login_name));
-            $this->user->fields['authtype'] = $authtype;
-            $user_dn                        = false;
+            $login_name                        = $this->user->fields['name'];
+            $this->auth_succeded               = true;
+            $this->user_present                = $this->user->getFromDBbyName(addslashes($login_name));
+            if (self::isAlternateAuth($authtype)) {
```

Comment:
This "MUST" be conditionalized, else it will be change for alternate auth, which is exactly what the PR want to fix.

       $this->user->fields['authtype'] = $authtype;